src/pages/rooms/chat-message.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import Chat from '../../types/chat';
+import marked from 'marked';
+import sanitizeHtml from 'sanitize-html';
+import emoji from 'node-emoji';
+import formatDistance from 'date-fns/formatDistance';
+import ChatInputForm from './chat-input-form';
+
+interface Props {
+  chat: Chat;
+  //room: Room,
+  setInput: Dispatch<SetStateAction<string>>;
+  onEdit: (chat: Chat, callback: Dispatch<SetStateAction<string>>) => void;
+  onCreate: (input: string, callback: Dispatch<SetStateAction<string>>) => void;
+}
+
+export default function ChatMessage({ chat, onCreate, onEdit, setInput }: Props) {
+  const [edit, setEdit] = useState<boolean>(false);
+  const [reply, setReply] = useState<boolean>(false);
+
+  const onClickEdit = (e: React.MouseEvent<HTMLAnchorElement>): void => {
+    setEdit(true);
+  };
+
+  const onClickReply = (e: React.MouseEvent<HTMLAnchorElement>): void => {
+    setReply(true)
+  };
+
+  const resetStatus = () => {
+    setReply(false);
+    setEdit(false);
+  };
+
+  const onCancel = (msg: string) => {
+    resetStatus();
+  }
+
+  const createReplyString = (chat: Chat, newInput: string): string => {
+    return `Reply to ${chat.user.name}'s Message on ${new Date(chat.updatedAt).toLocaleDateString()}: \n ${chat.msg.replace(/^/gm, "> ")} \n\n  ${newInput}`
+  }
```

Comment:
missing new line below

Reply:
If you meant below  / after function, then done